docs/docs/standard-lib/log.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: default
+title: Log
+nav_order: 16
+parent: Standard Library
+---
+
+# Log
+{: .no_toc }
+
+## Table of contents
+{: .no_toc .text-delta }
+
+1. TOC
+{:toc}
+
+---
+
+## Log
+
+To make use of the Log module an import is required.
+
+```cs
+import Log;
+```
+
+### Constants
+
+| Constant   | Description                                                           |
+| ---------- | --------------------------------------------------------------------- |
+| Log.stdout | The default file descriptor where a process can write output.         |
+| Log.stderr | An input stream where data is sent to and read by a program.          |
+| Log.stdin  | The default file descriptor where a process can write error messages. |
+
+### Log.print(string)
+
+```cs
+Log.print("something extremely interesting");
+```
+
+### Log.println(string)
+
+```cs
+Log.println("hello, world! println");
+```
+
+### Log.new(number)
+
+Create a new instance of a logger.
+
+```cs
+var log = Log.new(Log.stdout).unwrap();
+```
+
+### log.setPrefix(string)
+
+A prefix can be set on the logger that will be included in the output just before the user provided content.
+
+```cs
+log.setPrefix("prefix");
+```
+
+### log.unsetPrefix(string)
+
+Remove the prefix on the logger. This is a noop if there was no prefix previously set.
+
+```cs
+log.unsetPrefix();
```

Comment:
I wonder if you think this should also remove the prefixed timestamp that is appended to all logs (in case they want a custom time format for example)?

We can look at this as a future PR if you wish

Reply:
This is something I had considered but wasn't too sure about. I had modeled this after Go for the most part. I definitely think there's a lot of room for improvement where we could allow a user to set default fields, add fields, ad hoc, set custom timestamp formats, etc. A separate PR might be best to start investigating those ideas. However I'm open to your suggestions around that.

Reply:
Happy with investigating at a later date yep!